Task3/EEL_task_3/src/main/java/com/epam/esm/giftcertificate/service/GiftCertificateService.java

```diff
@@ -0,0 +1,102 @@
+package com.epam.esm.giftcertificate.service;
+
+import com.epam.esm.exceptionhandler.exceptions.NoSuchItemException;
+import com.epam.esm.exceptionhandler.exceptions.ObjectAlreadyExistsException;
+import com.epam.esm.exceptionhandler.exceptions.ObjectIsInvalidException;
+import com.epam.esm.giftcertificate.model.GiftCertificate;
+import com.epam.esm.giftcertificate.repository.GiftCertificateRepository;
+import com.epam.esm.tag.model.Tag;
+import com.epam.esm.tag.service.TagService;
+import com.epam.esm.utils.datavalidation.ParamsValidation;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class GiftCertificateService {
+    private final GiftCertificateRepository giftCertificateRepository;
+    private final TagService tagService;
+
+    @Autowired
+    public GiftCertificateService(GiftCertificateRepository giftCertificateRepository, TagService tagService) {
+        this.giftCertificateRepository = giftCertificateRepository;
+        this.tagService = tagService;
+    }
+
+    @Transactional
+    public boolean createCertificate(GiftCertificate giftCertificate) {
+        if (!giftCertificateRepository.isGiftCertificateExist(giftCertificate)) {
```

Comment:
Please re-write method, it's more readable when you throw exception in if statement first:
`if(exists) {
throw
}
if(!valid()) throw
doMainLogic
`

Reply:
Done